front/app/components/Form/Components/Layouts/CLSurveyPageLayout.tsx

```diff
@@ -225,24 +225,45 @@ const CLSurveyPageLayout = memo(
         return;
       }
 
-      console.log({ pageVariant });
-
       if (pageVariant === 'after-submission') {
         clHistory.push({ pathname: `/projects/${slug}` });
         return;
       }
 
       if (pageVariant === 'submission') {
         setIsLoading(true);
-        console.log('setting publication status:');
-        data.publication_status = 'published';
-        console.log({ data });
+        const dataWithPublicationStatus = {
+          ...data,
+          publication_status: 'published',
+        };
+
+        const idea: IIdea | undefined = await onSubmit(
+          { data: dataWithPublicationStatus },
```

Comment:
This was actually the bug. Before we were doing `onSubmit(data`, while we should be doing `onSubmit({ data `.

Reply:
The reason why it worked at all was because we fall back to some other internally maintained state in the `front/app/components/Form/index.tsx` component (the `const submissionData = formData && formData.data ? formData.data : data;` line). But this internally managed state didn't contain the `publication_status`. But apparently only on mobile or something.

Reply:
Anyway the lack of types is what really screwed us here. I've tried to improve it a little bit but there's only so much to be done without a refactor